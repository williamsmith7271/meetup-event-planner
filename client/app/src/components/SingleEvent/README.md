## SingleEvent Component
A component for showing a single event

### Example

```js
<SingleEvent />
```

### Other Information
See ./index.js for implementation details
