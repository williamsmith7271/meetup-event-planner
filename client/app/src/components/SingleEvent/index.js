import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Section from 'grommet-udacity/components/Section';
import Heading from 'grommet-udacity/components/Heading';
import Box from 'grommet-udacity/components/Box';
import Paragraph from 'grommet-udacity/components/Paragraph';
import Article from 'grommet-udacity/components/Article';
import CalendarIcon from 'grommet-udacity/components/icons/base/Calendar';
import LocationIcon from 'grommet/components/icons/base/Location';
import ClockIcon from 'grommet/components/icons/base/Clock';
import Footer from 'grommet/components/Footer';
import Button from 'grommet-udacity/components/Button';
import List from 'grommet/components/List';
import ListItem from 'grommet/components/ListItem';
import moment from 'moment';

const SingleEvent = ({
  event,
}) => (
  <Box align="center" justify="center">
    <Section justify="center">
      <Article className={styles.panel} pad={{ horizontal: 'small' }}>
        <Section>
          <Box pad={{ vertical: 'large' }} align="center">
            <CalendarIcon size="large" />
          </Box>
          <Heading align="center" tag="h3">
            {`It's a ${event.type} Party!`}
          </Heading>
          <Heading align="center">
            {event.name}
          </Heading>
          <Heading align="center" tag="h4">
            {`Hosted by ${event.host.name}`}
          </Heading>
        </Section>
        <Section>
          <Box justify="center" align="center" direction="column">
            <LocationIcon className={styles.icon} />
            <Heading tag="h4" align="center">
              {`${event.location}`}
            </Heading>
          </Box>
          <Box justify="center" align="center" direction="column">
            <ClockIcon className={styles.icon} />
            <Heading align="center" tag="h5">
              {`From: ${moment(event.start).format('LLL')}`}
            </Heading>
            <Heading align="center" tag="h5">
              {`To: ${moment(event.end).format('LLL')}`}
            </Heading>
          </Box>
        </Section>
        <Section>
          <Heading tag="h3" align="center">
            About
          </Heading>
          <Paragraph size="large" className={styles.paragraph}>
            {event.message}
          </Paragraph>
        </Section>
        <Section>
          <Heading tag="h3" align="center">
            Guest List
          </Heading>
          <Box justify="center" align="center" direction="column">
            <List>
              {event.guests.map((guest, i) =>
                <ListItem key={i}>
                  {guest.name}
                </ListItem>
              )}
            </List>
          </Box>
        </Section>
        <Footer align="center" justify="center" pad={{ vertical: 'medium' }}>
          <Button onClick={e => e} label="RSVP" />
        </Footer>
      </Article>
    </Section>
  </Box>
);

SingleEvent.propTypes = {

};

export default cssModules(SingleEvent, styles);
