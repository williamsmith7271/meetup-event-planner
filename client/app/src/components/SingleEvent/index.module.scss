.panel {
  background: #fff;
  box-shadow: 0 2px 4px 0 hsla(34,12%,61%,.52);
  width: 100%;
  margin-bottom: 80px;
  position: relative;
  min-width: 500px;
  @media screen and (max-width: 500px) {
    min-width: 300px;
    width: 100%;
  }
}

.icon {
  margin-top: 20px;
  margin-bottom: 10px;
}

.paragraph {
  padding: 0 40px;
}
