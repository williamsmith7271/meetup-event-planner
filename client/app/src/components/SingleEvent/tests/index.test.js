import SingleEvent from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<SingleEvent />', () => {
  it('should render with default props', () => {
    const event = {
      id: 4,
      name: 'Pool Party',
      end: '2014-12-12 00:00:00 UTC',
      start: '2016-11-02 00:00:00 UTC',
      type: 'birthday',
      location: '1204 North Carolina 55, Fuquay Varina, NC, United States',
      message: 'The best party',
      host: { name: 'Joaquin West' },
      guests: [{ name: 'Bilbo Baggins' }],
    };
    const user = {
      id: '1',
      bio: 'Experienced Software Engineer specializing in implementing cutting-edge technologies in a multitude of domains, focusing on Front End Web Development and UI / UX.',
      email: 'ryan@udacity.com',
      name: 'Ryan Collins',
      avatar: 'https://github.com/RyanCCollins/cdn/blob/master/misc/ryanc.jpg?raw=true',
      employer: 'Udacity',
      authToken: 'hzqmg5KWxJK7--4iWQYq',
      events: [{ name: 'Alumni App Release Party', id: '1' }]
    };
    const wrapper = shallow(
      <SingleEvent onRsvp={e => e} user={user} event={event} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
