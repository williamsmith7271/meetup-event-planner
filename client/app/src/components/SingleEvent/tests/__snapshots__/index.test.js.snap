exports[`<SingleEvent /> should render with default props 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  justify="center"
  pad="none"
  responsive={true}
  tag="div">
  <Section
    justify="center"
    pad={
      Object {
        "vertical": "medium"
      }
    }>
    <Article
      direction="column"
      pad={
        Object {
          "horizontal": "small"
        }
      }>
      <Section
        pad={
          Object {
            "vertical": "medium"
          }
        }>
        <Box
          align="center"
          announce={false}
          direction="column"
          focusable={true}
          pad={
            Object {
              "vertical": "large"
            }
          }
          responsive={true}
          tag="div">
          <Calendar
            a11yTitleId="calendar-title"
            responsive={true}
            size="large" />
        </Box>
        <Heading
          align="center"
          tag="h3">
          It\'s a birthday Party!
        </Heading>
        <Heading
          align="center"
          tag="h1">
          Pool Party
        </Heading>
        <Heading
          align="center"
          tag="h4">
          Hosted by Joaquin West
        </Heading>
      </Section>
      <Section
        pad={
          Object {
            "vertical": "medium"
          }
        }>
        <Box
          align="center"
          announce={false}
          direction="column"
          focusable={true}
          justify="center"
          pad="none"
          responsive={true}
          tag="div">
          <Location
            a11yTitleId="location-title"
            responsive={true} />
          <Heading
            align="center"
            tag="h4">
            1204 North Carolina 55, Fuquay Varina, NC, United States
          </Heading>
        </Box>
        <Box
          align="center"
          announce={false}
          direction="column"
          focusable={true}
          justify="center"
          pad="none"
          responsive={true}
          tag="div">
          <Clock
            a11yTitleId="clock-title"
            responsive={true} />
          <Heading
            align="center"
            tag="h5">
            From: November 2, 2016 12:00 AM
          </Heading>
          <Heading
            align="center"
            tag="h5">
            To: December 12, 2014 12:00 AM
          </Heading>
        </Box>
      </Section>
      <Section
        pad={
          Object {
            "vertical": "medium"
          }
        }>
        <Heading
          align="center"
          tag="h3">
          About
        </Heading>
        <Paragraph
          size="large">
          The best party
        </Paragraph>
      </Section>
      <Section
        pad={
          Object {
            "vertical": "medium"
          }
        }>
        <Heading
          align="center"
          tag="h3">
          Guest List
        </Heading>
        <Box
          align="center"
          announce={false}
          direction="column"
          focusable={true}
          justify="center"
          pad="none"
          responsive={true}
          tag="div">
          <List
            role="list">
            <ListItem
              align="center"
              direction="row"
              pad={
                Object {
                  "horizontal": "medium",
                  "vertical": "small"
                }
              }
              role="listitem"
              separator="bottom">
              Bilbo Baggins
            </ListItem>
          </List>
        </Box>
      </Section>
      <Footer
        align="center"
        direction="row"
        justify="center"
        pad={
          Object {
            "vertical": "medium"
          }
        }
        responsive={false}>
        <Button
          label="RSVP"
          onClick={[Function onRsvp]}
          type="button" />
      </Footer>
    </Article>
  </Section>
</Box>
`;
